packages/subgraph-service/contracts/utilities/AllocationManager.sol

```diff
@@ -378,8 +384,11 @@ abstract contract AllocationManager is EIP712Upgradeable, GraphDirectory, Alloca
         allocations[_allocationId].accRewardsPending += accRewardsPending;
 
         // Update total allocated tokens for the subgraph deployment
-        // underflow: subgraphAllocatedTokens should at least be oldTokens so it can't underflow
-        subgraphAllocatedTokens[allocation.subgraphDeploymentId] += (_tokens - oldTokens);
+        if (_tokens > oldTokens) {
```

Comment:
alternatively, this could be:
```
subgraphAllocatedTokens[allocation.subgraphDeploymentId] = subgraphAllocatedTokens[allocation.subgraphDeploymentId]  + _tokens - oldTokens;
```